src/ShopBundle/objects/TCMSSmartURLHandler/TCMSSmartURLHandler_BuyProductDirectLink.class.php

```diff
@@ -12,7 +12,10 @@
 /**
  * looks for URLS of the form /kaufen/ean/number... these will result in the article being
  * placed in the basket.
-/**/
+ *
+ * @psalm-suppress UndefinedPropertyAssignment
+ * @FIXME Writing data into `$OURLData` when there is no magic `__set` method for them defined.
```

Comment:
In this class, there is a lot of writing into 'magic' properties of `TCMSSmartURLData`  - however, those properties are not writable.